database-commons/src/main/java/io/cdap/plugin/db/connector/AbstractDBSpecificConnector.java

```diff
@@ -84,54 +87,102 @@ public InputFormatProvider getInputFormatProvider(ConnectorContext context, Samp
     ConnectionConfigAccessor connectionConfigAccessor = new ConnectionConfigAccessor();
     if (config.getUser() == null && config.getPassword() == null) {
       DBConfiguration.configureDB(connectionConfigAccessor.getConfiguration(), driverClass.getName(),
-                     getConnectionString(path.getDatabase()));
+                                  getConnectionString(path.getDatabase()));
     } else {
       DBConfiguration.configureDB(connectionConfigAccessor.getConfiguration(), driverClass.getName(),
-                     getConnectionString(path.getDatabase()), config.getUser(), config.getPassword());
+                                  getConnectionString(path.getDatabase()), config.getUser(), config.getPassword());
     }
-    String tableQuery = getTableQuery(path.getDatabase(), path.getSchema(), path.getTable(), request.getLimit());
+    // SessionID cannot be greater than 28 characters, as it is used as a suffix for variable names in the SQL queries
+    // in OracleConnector, and Oracle does not allow variable names over 30 characters.
```

Comment:
is this only a restriction for oracle? If so, it would be better to make a generateSessionID() method which has a default implementation, then have just the oracle plugin override it to some different logic.

It would also be a bit better to strip the '-' characters instead of replacing with '_', will provide 4 extra characters.